megatron/training.py

```diff
@@ -57,6 +57,49 @@
 from megatron.model.gpt2_model import cross_entropy
 from eval_tasks import run_eval_harness
 
+def save_base_shapes(neox_args, base_shapes, use_cache):
+    from copy import deepcopy
+    neox_args_copy = deepcopy(neox_args)
+
+    # Instantiation of the base model fails in the init function (init_functions.py) because we haven't called set_base_shapes on it at this point, so disable it temporarily here
+    neox_args.use_mup = False
+
+    base_model = GPT2ModelPipe(
+                    neox_args=neox_args,
+                    num_tokentypes=0,
+                    parallel_output=True,
+                    topology=mpu.get_topology(),
+                    use_cache=use_cache)
+
+    if not neox_args.is_pipe_parallel:
+        base_model = base_model.to_sequential()
+
+    base_shapes = mup.get_shapes(base_model)
+
+    del base_model
+
+    neox_args.hidden_size = neox_args.hidden_size * 2
+    neox_args.ffn_hidden_size = 4 * neox_args.hidden_size
```

Comment:
`ffn_hidden_size` and `kv_channels` don't exist in neox, so these two lines aren't doing anything.